"proj library not found" on F35/proj8 builds

```
checking for pj_init in -lproj... no
configure: error: proj library not found
error: Bad exit status from /var/tmp/rpm-tmp.BlJVHj (%build)
    Bad exit status from /var/tmp/rpm-tmp.BlJVHj (%build)
```

This happens on F35 and later builds, despite having proj-devel (8) installed

Full example log:
https://download.copr.fedorainfracloud.org/results/simc/stable/fedora-35-x86_64/03528692-fortrangis/builder-live.log.gz

--- a/configure.ac
+++ b/configure.ac
@@ -57,7 +57,7 @@ AC_ARG_ENABLE(proj, AC_HELP_STRING([--disable-proj],
 AM_CONDITIONAL(PROJ, test "x$enableval" = "xyes")
 
 if test "x$enableval" = "xyes"; then
-AC_CHECK_LIB([proj], [pj_init],
+AC_CHECK_LIB([proj], [proj_info],
  proj=yes
  PROJ_LIBS="-lproj"
  AC_SUBST(PROJ_LIBS)

--- a/configure.ac
+++ b/configure.ac
@@ -57,12 +57,17 @@ AC_ARG_ENABLE(proj, AC_HELP_STRING([--disable-proj],
 AM_CONDITIONAL(PROJ, test "x$enableval" = "xyes")
 
 if test "x$enableval" = "xyes"; then
-AC_CHECK_LIB([proj], [proj_info],
- proj=yes
- PROJ_LIBS="-lproj"
- AC_SUBST(PROJ_LIBS)
- AC_DEFINE(HAVE_LIBPROJ, 1, [Enable proj]),
- AC_MSG_ERROR([proj library not found]), )
+    AC_CHECK_LIB([proj], [pj_init], [proj=yes], [proj=no])
+    if test x$proj = xno; then
+        AC_CHECK_LIB([proj], [proj_info], [proj=yes], [proj=no])
+    fi
+    if test x$proj = xyes; then
+        PROJ_LIBS="-lproj"
+        AC_SUBST(PROJ_LIBS)
+        AC_DEFINE(HAVE_LIBPROJ, 1, [Enable proj])
+    else
+        AC_MSG_ERROR([proj library not found])
+    fi
 fi
 
 AC_ARG_ENABLE(shapelib, AC_HELP_STRING([--disable-shapelib],

